pyproject.toml

```diff
@@ -41,3 +41,12 @@ skip-magic-trailing-comma = false
 
 # Like Black, automatically detect the appropriate line ending.
 line-ending = "auto"
+
+[tool.pytest.ini_options]
+addopts = "--doctest-glob='**/*.md'"
+doctest_optionflags="NUMBER NORMALIZE_WHITESPACE ELLIPSIS"
+markers = [
+    "flash_attn_test: marks tests related to flash attention (deselect with '-m \"not flash_attn_test\"')",
+    "bitsandbytes: select (or deselect with `not`) bitsandbytes integration tests",
+    "generate: marks tests that use the GenerationTesterMixin"
+]
```

Comment:
What does that do exactly? Or does it come directly from Transformers?

Reply:
yeah, it comes from transformers https://github.com/huggingface/transformers/blob/f9c7e6021e9a9a9fd3fc8bb291da9451066aeb8d/pyproject.toml#L47-L54 and the markers are just tags for some of the tests.